### PR TITLE
Add basic usability check for type class members

### PR DESCRIPTION
**Pull request by LiamGoodacre**
For https://github.com/purescript/purescript/issues/2631

### CODE REVIEW

File: src/Language/PureScript/Errors.hs [hunk 1]
```diff
@@ -886,6 +887,11 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs) e = flip evalS
             , indent msg
             ]
 
+    renderSimpleErrorMessage (UnusableDeclaration ident) =
+      paras [ line $ "The declaration " <> markCode (showIdent ident) <> " is considered to be unusable."
```
Comment (paf31) [inline src/Language/PureScript/Errors.hs]: Nitpick: Is "considered to be" necessary?

File: examples/failing/UnusableTypeClassMethodSynonym.purs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-- @shouldFailWith UnusableDeclaration
+module Main where
+
+type M x = forall a. a
+
+class C a where
+  -- after synonym expansion, the type doesn't actually contain an `a`
+  c :: M a
```
Comment (paf31) [inline examples/failing/UnusableTypeClassMethodSynonym.purs]: There is a related problem: a type variable can be shadowed:

```purescript
c :: forall a. a -> a
```

has the same problem.

**Comment by paf31** [comment]
Thanks for working on this! Let's figure out a plan for dealing with shadowing before merging though. I think there might be an existing traversal we can use.

File: examples/failing/UnusableTypeClassMethodSynonym.purs (hunk 2 above)
Comment (LiamGoodacre) [inline examples/failing/UnusableTypeClassMethodSynonym.purs]: Have I misunderstood what `freeTypeVariables` does?  I was expecting to not get back any explicitly bound variables, such as those in introduced in a forall.

Comment (LiamGoodacre) [inline examples/failing/UnusableTypeClassMethodSynonym.purs]: In any case, I'll add a test case with a bound variable with the same name as a class argument.

Comment (paf31) [inline examples/failing/UnusableTypeClassMethodSynonym.purs]: Oh, you're right, that should do the job.

**Pull request merged**